### PR TITLE
Add support for deserializing source_mandate_notification objects

### PR DESCRIPTION
**Pull request by remi-stripe**
This fixes https://github.com/stripe/stripe-java/issues/445

r? @stan-stripe @ob-stripe 
cc @stripe/api-libraries

### CODE REVIEW

**Comment by spolu** [comment]
LGTM

File: src/main/java/com/stripe/model/SourceMandateNotification.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import com.stripe.exception.APIException;
+import com.stripe.exception.AuthenticationException;
+import com.stripe.exception.CardException;
+import com.stripe.exception.InvalidRequestException;
+import com.stripe.net.APIResource;
+import com.stripe.net.RequestOptions;
+
+
+public class SourceMandateNotification extends APIResource implements HasId, HasSourceTypeData {
+  String id;
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/SourceMandateNotification.java]: stripe-java uses tabs for indents.

File: src/main/java/com/stripe/net/APIResource.java [hunk 2]
```diff
@@ -61,6 +62,7 @@ public static void setStripeResponseGetter(StripeResponseGetter srg) {
 			.registerTypeAdapter(OrderItem.class, new OrderItemDeserializer())
 			.registerTypeAdapter(Source.class, new SourceTypeDataDeserializer<Source>())
 			.registerTypeAdapter(SourceTransaction.class, new SourceTypeDataDeserializer<SourceTransaction>())
+			.registerTypeAdapter(SourceMandateNotification.class, new SourceTypeDataDeserializer<SourceTransaction>())
```
Comment (ob-stripe) [inline src/main/java/com/stripe/net/APIResource.java]: Second argument should be `new SourceTypeDataDeserializer< SourceMandateNotification>`.

Comment (ob-stripe) [reply src/main/java/com/stripe/net/APIResource.java]: Also the list is in alphabetical order, so `SourceMandateNotification` should appear before `SourceTransaction`.

File: src/test/java/com/stripe/model/SourceMandateNotificationTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class SourceMandateNotificationTest extends BaseStripeTest {
+  @Before
+  public void mockStripeResponseGetter() {
+    APIResource.setStripeResponseGetter(networkMock);
```
Comment (ob-stripe) [inline src/test/java/com/stripe/model/SourceMandateNotificationTest.java]: You're not mocking any network requests in this class so I don't think you need this.

File: src/test/java/com/stripe/model/SourceMandateNotificationTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import com.stripe.net.RequestOptions.RequestOptionsBuilder;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class SourceMandateNotificationTest extends BaseStripeTest {
+  @Before
```
Comment (ob-stripe) [inline src/test/java/com/stripe/model/SourceMandateNotificationTest.java]: Use tabs here too.

File: src/main/java/com/stripe/net/APIResource.java (hunk 2 above)
Comment (remi-stripe) [reply src/main/java/com/stripe/net/APIResource.java]: @ob-stripe OOC why did this work then?

**Comment by remi-stripe** [comment]
@ob-stripe fixed the review comments, PTAL.

**Comment by ob-stripe** [comment]
You still need to register the new model in [`EventDataDeserializer`](https://github.com/stripe/stripe-java/blob/aad9e69f4d9e3e653d3f2f86f70c2010eb15894b/src/main/java/com/stripe/model/EventDataDeserializer.java#L56-L57).

**Comment by remi-stripe** [comment]
> You still need to register the new model in EventDataDeserializer.

How can I modify the test to catch that? I would have expected the tests to have failed in that case

**Comment by ob-stripe** [comment]
You'd need to deserialize a `source.mandate_notification` event object and make sure the nested object in `data.object` is correctly deserialized and cast to `SourceMandateNotification`.

**Comment by remi-stripe** [comment]
oh I see. Do you think it's worth adding?

I wonder how we could add an automated test to ensure new classes are properly listed in that method. I think I make similar mistakes in all our repos when I add a new resource.

**Comment by remi-stripe** [comment]
@ob-stripe PTAL after my new commit please!

**Pull request merged**